fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/BitmapValueIndexTest.java

```diff
@@ -483,9 +488,65 @@ public void nonOverlappingOrQuery() {
+    @Test
+    public void nestedAndQuery() {
+        final KeyExpression num_by_str = field("nested").nest(field("entry", FanOut).nest(concatenateFields("str_value", "num_value")));
+        final KeyExpression nested_num_by_str = concat(field("num_value_1"), num_by_str).group(1);
+        final KeyExpression nested_num_by_str_num2 = concat(field("num_value_1"), field("num_value_2"), num_by_str).group(1);
+        final KeyExpression nested_num_by_str_num3 = concat(field("num_value_1"), field("num_value_3"), num_by_str).group(1);
+        final RecordMetaDataHook nested_rec_no_by_str_nums_hook = metadata -> {
+            final RecordTypeBuilder recordType = metadata.getRecordType("MyNestedRecord");
+            metadata.addIndex(recordType, new Index("nested_num_by_str_num2", nested_num_by_str_num2, IndexTypes.BITMAP_VALUE, SMALL_BITMAP_OPTIONS));
+            metadata.addIndex(recordType, new Index("nested_num_by_str_num3", nested_num_by_str_num3, IndexTypes.BITMAP_VALUE, SMALL_BITMAP_OPTIONS));
+        };
+        final IndexAggregateFunction bitmap_value_nested_num_by_str = new IndexAggregateFunction(FunctionNames.BITMAP_VALUE, nested_num_by_str, null);
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, metaData(nested_rec_no_by_str_nums_hook));
+            for (int recNo = 100; recNo < 200; recNo++) {
+                recordStore.saveRecord(TestRecordsBitmapProto.MyNestedRecord.newBuilder()
+                        .setRecNo(recNo)
+                        .setNumValue1(1)
+                        .setNested(TestRecordsBitmapProto.MyNestedRecord.Nested.newBuilder()
+                                .addEntry(TestRecordsBitmapProto.MyNestedRecord.Nested.Entry.newBuilder()
+                                        .setStrValue((recNo & 1) == 1 ? "odd" : "even")
+                                        .setNumValue(recNo + 1000)))
+                        .setNumValue2(recNo % 7)
+                        .setNumValue3(recNo % 5)
+                        .build());
+            }
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, metaData(nested_rec_no_by_str_nums_hook));
+            setupPlanner(null);
+            final RecordQuery recordQuery = RecordQuery.newBuilder()
+                    .setRecordType("MyNestedRecord")
+                    .setFilter(Query.and(
+                            Query.field("num_value_1").equalsValue(1),
+                            Query.field("nested").matches(Query.field("entry").oneOfThem().matches(Query.field("str_value").equalsValue("odd"))),
+                            Query.field("num_value_2").equalsValue(3),
+                            Query.field("num_value_3").equalsValue(4)))
+                    .setRequiredResults(Collections.singletonList(field("nested").nest(field("entry", FanOut).nest("num_value"))))
```

Comment:
1. Adding `str_value` is fine; the partial record has an entry with both -- which doesn't matter much, since we just get the index entry directly.
2. Removing the condition and expecting a rollup is sensible, but does not work with the code today. Specifically, it fails `separateGroupFilters` as not matching the `bitmap_value` operand.

Reply:
Okay, sounds good. I suppose at some point, we may care about roll up, but that sounds fine for now.